pedsnetdcc/prepdb.py

```diff
@@ -0,0 +1,244 @@
+create schema if not exists {{.Site}}_id_maps  authorization dcc_owner;
+grant usage  on               schema {{.Site}}_pedsnet  to harvest_user, achilles_user, dqa_user, pcor_et_user, peds_staff;
+grant select on all tables in schema {{.Site}}_pedsnet  to harvest_user, achilles_user, dqa_user, pcor_et_user, peds_staff;
+grant all    on               schema {{.Site}}_pedsnet  to loading_user;
+grant all    on               schema {{.Site}}_pcornet  to pcor_et_user;
+grant usage  on               schema {{.Site}}_pcornet  to peds_staff;
+grant select on all tables in schema {{.Site}}_pcornet  to peds_staff;
+grant all    on               schema {{.Site}}_harvest  to harvest_user;
+grant all    on               schema {{.Site}}_achilles to achilles_user;
+grant all    on               schema {{.Site}}_id_maps  to loading_user;
+alter default privileges for role loading_user in schema {{.Site}}_pedsnet grant select on tables to harvest_user, achilles_user, dqa_user, pcor_et_user, peds_staff;
+alter default privileges for role loading_user in schema {{.Site}}_pcornet grant select on tables to peds_staff;
+"""
+
+# SQL template for creating site schemas in a production database instance.
+_sql_site_prod_template = """
+create schema if not exists {{.Site}}_pedsnet authorization dcc_owner;
+create schema if not exists {{.Site}}_pcornet authorization dcc_owner;
+create schema if not exists {{.Site}}_harvest authorization dcc_owner;
+create schema if not exists {{.Site}}_id_maps  authorization dcc_owner;
+grant usage  on               schema {{.Site}}_pedsnet  to harvest_user, peds_staff;
+grant select on all tables in schema {{.Site}}_pedsnet  to harvest_user, peds_staff;
+grant usage  on               schema {{.Site}}_pcornet  to pcornet_sas, peds_staff;
+grant select on all tables in schema {{.Site}}_pcornet  to pcornet_sas, peds_staff;
+grant all    on               schema {{.Site}}_harvest  to harvest_user;
+grant all    on               schema {{.Site}}_id_maps  to loading_user;
+alter default privileges for role dcc_owner in schema {{.Site}}_pedsnet grant select on tables to harvest_user, peds_staff;
+alter default privileges for role dcc_owner in schema {{.Site}}_pcornet grant select on tables to pcornet_sas, peds_staff;
+"""
+
+
+def _site_sql(site, db_type):
+    """Return a list of statements to create the schemas for a given site.
+
+    :param site: site name, e.g. 'dcc' or 'stlouis'
+    :type: str
+    :param db_type: 'prod' or 'internal'
+    :type: str
+    :return: SQL statements
+    :rtype: list(str)
+    :raises: ValueError
+    """
+    if db_type == 'internal':
+        tmpl = _sql_site_internal_template
+    elif db_type == 'prod':
+        tmpl = _sql_site_prod_template
+    else:
+        raise ValueError('Invalid db_type: {}'.format(db_type))
+
+    sql = tmpl.replace('{{.Site}}', site)
+    return [x for x in sql.split("\n") if x]
+
+
+_sql_vocab_internal = """
```

Comment:
We also need a `dcc_ids` schema to hold the `last_id` value for each domain during the id mapping (see [here](https://github.com/PEDSnet/dcc-loader/blob/master/id-map/create-all-schemas.sql#L135)). The pertinent statements are:

```
create schema if not exists dcc_ids authorization dcc_owner;
grant all on schema dcc_ids to loading_user;
```

I don't think anyone else besides the `loading_user` will need access to this schema. We could shove it into the vocab sql lists, since that only gets executed once, or make a separate set for it. Up to you.